mem.go

```diff
@@ -198,18 +198,19 @@ func (m *MemoryHandler) Find(ctx context.Context, lookup *resource.Lookup, page,
 		}
 		// Apply pagination
 		total := len(items)
-		start := (page - 1) * perPage
 		end := total
-		if perPage > 0 {
-			end = start + perPage
+		start := offset
+
+		if limit > 0 {
+			end = start + limit
 			if start > total-1 {
 				start = 0
 				end = 0
 			} else if end > total-1 {
 				end = total
 			}
 		}
-		list = &resource.ItemList{total, page, items[start:end]}
+		list = &resource.ItemList{Total: total, Page: offset, Offset: offset, Items: items[start:end]}
```

Comment:
I would rename `Offset` to `Skip` and use `skip` query-string parameter instead of `offset`.